[MRG]  add transform y section to faq.rst

theoptips opened the pull request:
Add transform y in a pipeline Q and A to FAQ.

#### Reference Issues/PRs
Fixes  #12587

#### What does this implement/fix? Explain your changes.
Add transform y section, related discussion, suggested libraries to FAQ

#### Any other comments?
N/A

Commit:
Github issue #12587 add to faq.rst transform y discussion, alternative extensions

theoptips committed:
Change URL formatting

theoptips committed:
Change URL format

thomasjpfan commented on doc/faq.rst:
```diff
@@ -385,3 +385,7 @@ efficient to process for most operations. Extensive work would also be needed
+
+Do you plan to implement transform for target y in a pipeline? Currently transform only works for features X in a pipeline.
+----------------------------------------------------------------------------
+This a long-standing discussion about not being able to transform y in a pipeline, follow on github issue #4143. Meanwhile check out TargetTransformer, pipegraph, `imbalanced-learn<https://github.com/scikit-learn-contrib/imbalanced-learn>`
```
1. Please make this <80 lines to match the convention in the file.

2. Are you refering to `TransformedTargetRegressor` here?

3. Link to pipegraph?

theoptips replied:
@thomasjpfan thanks for reviewing.
1. Do you mean the title? Edited to just one line
2. Confirmed with amueller and changed to TransformedTargetRegressor 
3. Confirmed with amueller adn linked to pipegraph

theoptips replied:
@thomasjpfan 
1. Fixed line length
2. Fixed. Style improved
3. Fixed

thomasjpfan reviewed (commented):
Thank you for the PR @theoptips !

theoptips committed:
Add format link

Changed to TransformedTargetRegressor
Re-formatted transform y section
Added links to pipegraph imbalanced-learn

TomDLT commented on doc/faq.rst:
```diff
@@ -385,3 +385,7 @@ efficient to process for most operations. Extensive work would also be needed
+
+Do you plan to implement transform for target y in a pipeline?
+----------------------------------------------------------------------------
+Currently transform only works for features X in a pipeline. There's a long-standing discussion about not being able to transform y in a pipeline. Follow on github issue #4143. Meanwhile check out sklearn.compose.TransformedTargetRegressor, `pipegraph<https://github.com/mcasl/PipeGraph>`_, `imbalanced-learn<https://github.com/scikit-learn-contrib/imbalanced-learn>`_
```
We like to keep every line under 80 characters, so you have to split this long line into multiple lines

```suggestion
Currently transform only works for features X in a pipeline. There's a
long-standing discussion about not being able to transform y in a pipeline.
Follow on github issue #4143. Meanwhile check out
sklearn.compose.TransformedTargetRegressor,
`pipegraph<https://github.com/mcasl/PipeGraph>`_,
`imbalanced-learn<https://github.com/scikit-learn-contrib/imbalanced-learn>`_
```

TomDLT replied:
Other comments:
- Can you add a link on `#4143` to be able to go directly to this discussion?
- You can change `sklearn.compose.TransformedTargetRegressor` into
```py
:class:`sklearn.compose.TransformedTargetRegressor`
```
which will automatically link to the corresponding page in the documentation.

theoptips replied:
@TomDLT Changes made.

theoptips committed:
Update doc/faq.rst

styling change

Co-Authored-By: Tom Dupré la Tour <tom.dupre-la-tour@m4x.org>

theoptips committed:
Update faq.rst

:class:`sklearn.compose.TransformedTargetRegressor`

TomDLT commented on doc/faq.rst:
```diff
@@ -385,3 +385,12 @@ efficient to process for most operations. Extensive work would also be needed
+----------------------------------------------------------------------------
+Currently transform only works for features X in a pipeline. There's a
+long-standing discussion about not being able to transform y in a pipeline.
+Follow on github issue #4143. Meanwhile check out
```
Could you add the link to the discussion:
```
`#4143<https://github.com/scikit-learn/scikit-learn/issues/4143>`
```
You may need to rearrange the lines to stay below the 80 characters per line limit.

theoptips replied:
Added link to discussion 4143

jnothman commented:
I think this should include a bit more of the discussion. Note that we have solved for the case where y has an invertible transformation applied before training and inverted after prediction... And we hope to solve for use cases where y should be transformed at training time and not at test time, for resampling and similar uses, like at imbalanced learn. In general these use cases should be solved with a custom meta estimator rather than a Pipeline

theoptips committed:
Styling: link to issue

Link to 4143 issue

theoptips committed:
Added additional discussion

Added discussion from jnothman

theoptips committed:
Updated faq.rst

updated language so it's more of a suggestion

cmarmo commented:
@jnothman , @TomDLT , @thomasjpfan had all comments be addressed here? Thanks!

TomDLT commented:
Thanks @theoptips  !
